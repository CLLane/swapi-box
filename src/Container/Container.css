.card-container {
  overflow-y: scroll;
  height: 650px;
  margin-top: 20px;
}
