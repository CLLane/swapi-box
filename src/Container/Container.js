import React from 'react';
import Card from '../Card/Card.js'
import './Container.css';

const Container = ({people, planets, vehicles, films}) => {

  const peopleInfo = people.map(person => {
    return <Card key={person.name + Date.now()} name={person.name} homeworld={person.homeworld} species={person.species} population={person.population}/>
  })

  const planetsInfo = planets.map(planet => {
    return <Card key={planet.name + Date.now()} name={planet.name} terrain={planet.terrain} population={planet.population} climate={planet.climate} residents={planet.residents}/>
  })

  const vehiclesInfo = vehicles.map(vehicle => {
    return <Card key={vehicle.name + Date.now()} name={vehicle.name} model={vehicle.model} vehicleClass={vehicle.vehicleClass} numberOfPassengers={vehicle.numberOfPassengers}/>
  })

  return (
    <div>
      <section>{peopleInfo}</section>
      <section>{planetsInfo}</section>
      <section>{vehiclesInfo}</section>
    </div>
  )
}

export default Container;