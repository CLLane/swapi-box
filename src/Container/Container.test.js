import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const data = [{
      id: 1,
      name: 'Luke Skywalker',
      homeworld: 'Tatooine',
      species: 'Human',
      population: 200000
    }]
    const wrapper = shallow(<Container
      data= {data}
      toggleFav={jest.fn()}
      type = '/people'
      />);

      expect(wrapper).toMatchSnapshot(); 
  })
});