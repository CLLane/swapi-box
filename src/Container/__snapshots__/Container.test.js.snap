// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot with all data passed in correctly 1`] = `
<div>
  <section
    className="card-container"
  >
    <NavLink
      key="1"
      to="//people/1"
    >
      <Card
        homeworld="Tatooine"
        id={1}
        key="Luke Skywalker1566876964528"
        name="Luke Skywalker"
        population={200000}
        species="Human"
        toggleFav={[MockFunction]}
        type="/people"
      />
    </NavLink>
  </section>
</div>
`;
